test/box-luatest/gh_9823_net_replicaset_test.lua

```diff
@@ -0,0 +1,161 @@
+local t = require('luatest')
+local cbuilder = require('test.config-luatest.cbuilder')
+local cluster = require('test.config-luatest.cluster')
+
+local g = t.group()
+
+g.before_all(cluster.init)
+g.after_all(cluster.clean)
+g.after_each(cluster.drop)
+
+local function test_config()
+    local sharding_role = {
+        privileges = {{permissions = {'execute'}, universe = true}},
+    }
+    local builder = cbuilder.new()
+    :set_global_option('credentials.roles.sharding', sharding_role)
+    :set_global_option('credentials.users.storage.roles', {'sharding'})
+    :set_global_option('credentials.users.storage.password', 'secret_storage')
+    :set_global_option('iproto.advertise.sharding.login', 'storage')
+    :use_group('test')
+    builder:use_replicaset('router_1')
+           :add_instance('router_1', {})
+    builder:use_replicaset('storage_a')
+            :set_replicaset_option('replication.failover', 'election')
+    for i = 1, 3 do
+        builder:add_instance('storage_a' .. i, {})
+    end
+    return builder:config()
+end
+
+-- Test errors of replicaset connector.
+g.test_net_replicaset_error = function(cg)
+    local config = test_config()
+    config.groups.test.replicasets.storage_a.replication.failover = 'supervised'
+    local cluster = cluster.new(cg, config)
+    cluster:start()
+
+    -- Check unknown replicaset does not work.
+    cluster.router_1:exec(function()
+        local t = require('luatest')
+        local net_replicaset = require('internal.net.replicaset')
+        local err_msg = 'The replicaset was not found by its name'
+        t.assert_error_msg_equals(err_msg, net_replicaset.connect, 'moon')
+    end)
+
+    -- Make all instances read-only.
+    cluster:each(function(server)
+        if server.alias:startswith('storage') then
+            server:exec(function()
+                box.cfg{read_only = true}
+            end)
+        end
+    end)
+
+    -- Check that the replicaset does not work.
+    cluster.router_1:exec(function()
+        local t = require('luatest')
```

Comment:
From what I know, you don't need to do it in execs anymore. It is there by default with the same name as in the test file.